### PR TITLE
Order List Management Step 1: Basic Order List Support + MVVM

### LINKED ISSUE #1330: Hack Week: Convert Order List to use new list management backend
Convert the order list tab to use the new order list management implemented in FluxC.

## TODO

- [x] Basic list conversion and MVVM conversion (#1496 ) 
- [x] Hide future orders (#1497, [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1410), #1510)
- [x] Offline handling (#1495, PR #1513, [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1412))
- [x] Error, Loading & Empty States (#1514, PR #1530, [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1419))
- [x] Fix tests (#1533, PR #1538  )

### PR DESCRIPTION
**Pull request by AmandaRiu**
Partially fixes #1330

## Included in this PR

* Orders List code updated to use `MVVM`
* Orders are now fetched and managed using the new order list management logic
* New loading skeleton implementation
* Load more
* Pull to refresh
* Search and Filter
* Click to open an order
* Updating an order status will cause the order list to refresh

#### Order List Loading

orders-list-loading

## Not Included

The following items will be handled in a future PR:
* Removing future dated orders (needs to be done in FluxC)
* Offline handling
* Error, loading & empty states
* Unit and instrumentation test updates

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ScopedAndroidViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Application context aware {@link ScopedViewModel}
+ * <p>
+ * Subclasses must have a constructor which accepts {@link Application} as the only parameter.
+ * <p>
+ */
+@SuppressLint("StaticFieldLeak")
+abstract class ScopedAndroidViewModel(
+    defaultDispatcher: CoroutineDispatcher,
+    val appContext: Application
+) : ScopedViewModel(defaultDispatcher)
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ScopedAndroidViewModel.kt]: This doesn't seem to be used anywhere at the moment?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/ThrottleLiveData.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * <p>
+ * This class correctly propagates its active/inactive states down to source {@code LiveData}
+ * objects after a set delay. There can only be one event delayed at a time. If an event is already delayed when
+ * a new one is received, then the previous Job to propagate the event down to the observers is canceled.
+ * <p>
+ *
+ * @param <T> The type of data hold by this instance
+ */
+class ThrottleLiveData<T> constructor(
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/ThrottleLiveData.kt]: TIL 😄

File: WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ScopedAndroidViewModel.kt (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ScopedAndroidViewModel.kt]: Ah yes, a holdover from my old implementation. Removed in 08937357b9a41f9fea32a400fbf7454712bd5cfe

**Comment by AmandaRiu** [comment]
@anitaa1990 Thank you for the review! I've fixed the crash in c8e5d784d4405eaced64314258952828fc5edc64, and addressed the code comments as well. Ready for another round.

**Pull request merged**